Added set TTS lanuage in context reviewer menu

arooooon opened the pull request:
I hope everything is fine, at least it works when I tried it. Would be good if someone could take a look at my sql query in MetaDb.. wouldn't wanna mess up the database ;)

This will ask the user to select language for the current cardside.


Also I suggest we change:
"Reset language assignments (for text to speech and dictionaries) for all decks. Reset specific deck assignments on deck picker."

To
"Reset language assignments (for text to speech and dictionaries) for all decks. Reset specific deck assignments in reviewer."

timrae commented on AnkiDroid/src/main/res/values/10-preferences.xml:
```diff
@@ -54,7 +54,7 @@
     <string name="dictionary">Lookup dictionary</string>
     <string name="reset_languages">Reset languages</string>
-    <string name="reset_languages_summ">Reset language assignments (for text to speech and dictionaries) for all decks. Reset specific deck assignments on deck picker.</string>
+    <string name="reset_languages_summ">Reset language assignments (for text to speech and dictionaries) for all decks. Reset specific deck assignments in reviewer.</string>
```
I think it's better to just remove the 'Reset specific deck assignments in reviewer.' part... We can add a note about that to the manual.

timrae commented on AnkiDroid/src/main/java/com/ichi2/anki/ReadText.java:
```diff
@@ -74,31 +74,19 @@ public static String getLanguage(long did, int ord, int qa) {
-        String language = getLanguage(mDid, mOrd, mQuestionAnswer);
-        Timber.d("ReadText.textToSpeech() method found language choice '%s'", language);
-        // rebuild the language list if it's empty
-        if (availableTtsLocales.isEmpty()) {
```
With this change it looks like `selectTts()` won't work unless `textToSpeech()` is called before it, since that calls `buildAvailableLanguages()`... Is that intentional?

timrae commented:
Other than those things, it looks good! When you've finished everything would you be able to [squash it down to one commit](http://stackoverflow.com/questions/5189560/squash-my-last-x-commits-together-using-git) please? You'll need to do a force push after that to update the current PR.

arooooon commented:
> I think it's better to just remove the 'Reset specific deck assignments in reviewer.' part...

Yes I removed that text.

> With this change it looks like selectTts() won't work properly unless textToSpeech() is called before it, since that calls buildAvailableLanguages()... Is that intentional?

No, thanks that could have become a bug in some rare cases. I added buildAvailableLanguages() to selectTts() as well, hope that works.

timrae commented:
Thank you for working on this, it's a useful addition!
